doc/designs/quic-design/quic-ackm.md

```diff
@@ -0,0 +1,411 @@
+
+The structure pointed to by the returned pointer, and the referenced ACK range
+structures, are guaranteed to remain valid until the next call to any
+`QUIC_ACKM` function. After such a call is made, all fields become undefined.
+
+This function is used to provide ACK frames for acknowledging packets which have
+been received and notified to the ACK manager via `QUIC_ACKM_on_rx_packet`.
+
+```c
+const QUIC_ACKM_ACK *QUIC_ACKM_get_ack_frame(QUIC_ACKM *ackm);
+```
+
+### Get Probe Packet
+
+This determines if the ACK manager is requesting any probe packets to be
+transmitted.
+
+The user calls `QUIC_ACKM_get_probe_request`. The structure pointed
+to by `info` is filled and the function returns 1 on success.
+
+The fields of `QUIC_ACKM_PROBE_INFO` record the number of probe requests
+of each type which are outstanding. In short:
+
+  - `handshake` designates the number of ACK-eliciting Handshake
+    packets being requested. This is equivalent to
+    `SendOneAckElicitingHandshakePacket()` in RFC 9002.
+
+  - `padded_initial` designates the number of ACK-eliciting
+    padded Initial packets being requested. This is equivalent to
+    `SendOneAckElicitingPaddedInitialPacket()` in RFC 9002.
+
+  - `pto` designates the number of ACK-eliciting outstanding probe events
+    corresponding to each packet number space. This is equivalent to
+    `SendOneOrTwoAckElicitingPackets(pn_space)` in RFC 9002.
+
+Once the user has processed these requests, the user must clear these
+outstanding requests by calling `QUIC_ACKM_get_probe_request` with `clear` set
+to 1. If `clear` is non-zero, the current values are returned and then zeroed,
+so that the next call to `QUIC_ACKM_get_probe_request` (if made immediately)
+will return zero values for all fields.
+
+```c
+typedef struct quic_ackm_probe_info_st {
+    uint32_t handshake, padded_initial, pto[QUIC_PN_SPACE_NUM];
+} QUIC_ACKM_PROBE_INFO;
+
+int QUIC_ACKM_get_probe_request(QUIC_ACKM *ackm, int clear,
+                                QUIC_ACKM_PROBE_INFO *info);
+```
+
+### Get Largest Unacked Packet Number
+
+This gets the largest unacknowledged packet number in the given packet number
+space. The packet number is written to `*pn`. Returns 1 on success.
+
+This is needed so that packet encoders can determine with what length to encode
+the abridged packet number in the packet header.
+
+```c
+int QUIC_ACKM_get_largest_unacked(QUIC_ACKM *ackm, int pkt_space, QUIC_PN *pn);
+```
+
+RTT Tracker
+-----------
+
+RTT statistics must be kept and will be needed by the ACK manager, but also are
+likely to be needed by other parts of the QUIC implementation.
+
+The ACK manager consumes the following interface (which could be polymorphic or
+have a fixed implementation):
+
+```c
+typedef struct quic_rtt_stats_st {
+    /* As defined in RFC 9002. */
+    OSSL_TIME min_rtt,
+              latest_rtt,
+              smoothed_rtt,
+              max_ack_delay;
+} QUIC_RTT_STATS;
```

Comment:
Wouldn't these be part of the flow controller/statistics collector?

More will be required for congestion control from memory (after MVP).